### ISSUE TITLE
php8.4 support

### ISSUE THREAD

**Issue by jadeops** [tamtamchik/simple-flash#17]
**Describe the bug**
deprecated messages in logs. there could be more

```
PHP Deprecated: Tamtamchik\SimpleFlash\Flash::__construct(): Implicitly marking parameter $template as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Flash.php on line 50
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::display(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 69
PHP Deprecated: Tamtamchik\SimpleFlash\Core\MessageManager::hasMessage(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/MessageManager.php on line 90
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::isReadyForDisplay(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 100
PHP Deprecated: Tamtamchik\SimpleFlash\Core\MessageManager::clearMessages(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/MessageManager.php on line 112
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::some(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 116
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::clear(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 128
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displayBootstrap(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 210
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displayFoundation(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 222
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displayBulma(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 234
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displayMaterialize(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 246
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displayTailwind(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 258
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displayPrimer(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 270
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displayUiKit(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 282
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displaySemantic(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 294
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displaySpectre(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 306
PHP Deprecated: Tamtamchik\SimpleFlash\Core\Engine::displayHalfmoon(): Implicitly marking parameter $type as nullable is deprecated, the explicit nullable type must be used instead in /vendor/tamtamchik/simple-flash/src/Core/Engine.php on line 318

```

**Expected behavior**
no deprecated messages in logs

**Comment by ToshY** [comment]
Also stumbled across this when trying to upgrade to PHP 8.4.1. @tamtamchik any chance you have some time to take a look at this?

**Comment by tamtamchik** [comment]
@jadeops @ToshY Will take a look on the weekend!

**Comment by tamtamchik** [comment]
@ToshY @jadeops fixed in [3.0.1](https://github.com/tamtamchik/simple-flash/releases/tag/3.0.1)

**Comment by ToshY** [comment]
> @ToshY @jadeops fixed in [3.0.1](https://github.com/tamtamchik/simple-flash/releases/tag/3.0.1)

Much appreciated! 🥇
